controllers/cart.ts

```diff
@@ -45,13 +51,17 @@ export const calculateCartController = buildEndpointRegister({
   handler: (data) => {
     const items = data.cart.lineItems;
 
+    // PRO: Great strategy! breaking the problem into two "parts".
     const [withDiscount, withoutDiscount] = partition(
       isDiscountableItem,
       items
     );
 
     const discount = getDiscountPercentage(withDiscount.length);
 
+    // This is more like a personal preference, but I think it would easier to read
```

Comment:
Agree! There's no need to duplicate the calculations here since they're already happening below. `total` should be the result of summing `finalPrice` from each item in the resulted list